superset-frontend/src/features/alerts/components/NotificationMethod.tsx

```diff
@@ -437,7 +437,7 @@ export const NotificationMethod: FunctionComponent<NotificationMethodProps> = ({
             />
             {index !== 0 && !!onRemove ? (
               <span
-                role="button"
+                role="delbutton"
```

Comment:
`role` attribute should only use [aria roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) since the application's accessibility depends on it.

Reply:
Please could you suggest , can I change `role` with `data-testid`

like below

```
               <span
                  className="ghost-button"
                  data-testid="ccbutton"
                  tabIndex={0}
                  onClick={() => setCcVisible(true)}
                  style={{ display: ccVisible ? 'none' : 'inline-flex' }}
                >
```

```
              <span
                data-testid="delbutton"
                tabIndex={0}
                className="delete-button"
                onClick={() => onRemove(index)}
              >

```

```
                <span
                  className="ghost-button"
                  data-testid="bccbutton"
                  tabIndex={0}
                  onClick={() => setBccVisible(true)}
                  style={{ display: bccVisible ? 'none' : 'inline-flex' }}
                >
```

and relevant test case to evaluate real delete is as below

```
const deleteButton = screen.getByTestId('delbutton');
userEvent.click(deleteButton);

```

Reply:
Thanks @kgabryje !  Updated test case and validated to be success.

```
npm run test -- --silent src/features/alerts/components/NotificationMethod.test.tsx

 PASS  src/features/alerts/components/NotificationMethod.test.tsx

Test Suites: 1 passed, 1 total
Tests:       7 passed, 7 total
Snapshots:   0 total
Time:        4.149 s, estimated 7 s

```

Reply:
@kgabryje 

The "Python-Integration / test-mysql (pull_request)" job is failing intermittently and appears to be flaky. A retrigger or reattempt should succeed.

Reply:
yeah its definitely flaky... I re-run it, hopefully it passes now

Reply:
Thanks again, @kgabryje! All tests are now successful. Please assist with merging this PR.

Reply:
Thanks again, @kgabryje! All tests are now successful. Please assist with merging this PR.